src/multivariate/mvnormal.jl

```diff
@@ -142,7 +142,7 @@ sqmahal(d::AbstractMvNormal, x::AbstractMatrix) = sqmahal!(Vector{promote_type(p
 
 _logpdf(d::AbstractMvNormal, x::AbstractVector) = mvnormal_c0(d) - sqmahal(d, x)/2
 
-function _logpdf!(r::AbstractArray, d::AbstractMvNormal, x::AbstractMatrix)
+function _logpdf!(r::AbstractVector{<:Real}, d::AbstractMvNormal, x::AbstractMatrix{<:Real})
```

Comment:
Does this introduce a new restriction?

Reply:
Ah it seems we currently allow outputs of any shape and only check if the length is correct. I'll change this, also in the `logpdf!` definition.

BTW maybe a not completely obvious consequence of this PR is that it would be possible to remove the non-exported `_logpdf` and `_rand!` functions from the interface. Currently we call them if we want to avoid bounds checking but this leads to problems if users or developers implement `logpdf` or `rand!` instead (sometimes one wants to use more efficient implementations for the matrix case and then it is easier to just implement a single `logpdf` function for matrices and vectors; this is done e.g. in AbstractGPs). With this PR, however, we only call `logpdf` and `rand!` internally but use `@inbounds` to indicate that bounds checks should be elided if we are certain that the sizes are correct.

Reply:
I fixed the inplace methods.